pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -675,21 +784,29 @@ func (nc *Controller) doEvictionPass() {
 				klog.Warningf("Failed to get Node %v from the nodeLister: %v", value.Value, err)
 			}
 			nodeUID, _ := value.UID.(string)
-			pods, err := listPodsFromNode(nc.kubeClient, value.Value)
-			if err != nil {
-				utilruntime.HandleError(fmt.Errorf("unable to list pods from node %q: %v", value.Value, err))
-				return false, 0
+			pods, ok := nc.nodeEvictionMap.deletePods(value.Value)
```

Comment:
I should have done this thinking some time ago, but I started wondering if this whole PR isn't a bit of overkill.
I need to think a bit more about that, but at the high level what I'm thinking about is:
- assuming we're listing pods from cache, and ignoring NotFound errors in evicting and marking pods as not ready, it seems completely fine to me to avoid passing pods around in this code and just simply rely on what is in cache
- we are only risking that we will try to evict the same Pod N times, which isn't correctness issue and in general this should happen extremely rare...

So i guess what I'm suggesting is is roughly:
- remove references to pods from nodeEvictionMap completely - just keep the state
- here just list pods from cache
- modify the logic in evictPods() to list pods from node in the "evicted" case - from cache - we have guarantee that the cache state is at least synced to the point of handler when handler is called
- handle retries very similarly (or exactly as now?) in processPod()

Reply:
I thought about it a bit more, and I'm fairly confident that it would be fine (and obviously much simpler).
It requires adding comments in MarkPodsNotReady and DeletePods to ignore NotFound errors...